deps: [M3-9471] - Clean up unnecessary package resolutions

bill-akamai opened the pull request:
## Description 📝

This PR removes unnecessary package resolutions from the root `package.json`. After checking each resolution with `pnpm why -r`, the `cookie` and `nanoid` packages are now naturally resolving to newer versions. `node-fetch`, `yaml`, and `semver` still use dependencies that rely on outdated versions, so those resolutions are being retained for now.

## 🗑️ Resolutions removed:

### **`cookie`** pinned version: `^0.7.0`
Packages are resolving to `0.7.2`

**Relevant PR(s)**:
  - https://github.com/linode/manager/pull/11434

**Dependabot alert(s)**:
  - https://github.com/linode/manager/security/dependabot/131

**Jira ticket(s)**:
  - https://jira.linode.com/browse/M3-8782

---

### **`nanoid`** pinned version: `^3.3.8`
Packages are resolving to `3.3.11`

**Relevant PR(s)**:
  - https://github.com/linode/manager/pull/11510

**Dependabot alert(s)**:
  - https://github.com/linode/manager/security/dependabot/137

**Jira ticket(s)**:
  - https://jira.linode.com/browse/M3-9083

---

## 📌 Resolutions still needed:

### **`node-fetch`** pinned version: `^2.6.7`
```
recompose 0.30.0
└─┬ fbjs 0.8.18
  └─┬ isomorphic-fetch 2.2.1
    └── node-fetch 1.7.3
```

**Relevant PR(s)**:
  - https://github.com/linode/manager/pull/7014
  - https://github.com/linode/manager/pull/8696

**Dependabot alert(s)**:
  - There's been several over the years: https://github.com/linode/manager/pulls?q=is%3Apr+is%3Aclosed+author%3Aapp%2Fdependabot+node-fetch

**Jira ticket(s)**:
  - No associated Jira tickets found

---

### **`yaml`** pinned version: `^2.3.0`
```
@emotion/react 11.13.5
└─┬ @emotion/babel-plugin 11.13.5
  └─┬ babel-plugin-macros 3.1.0
    └─┬ cosmiconfig 7.1.0
      └── yaml 1.10.2
```

**Relevant PR(s)**:
  - https://github.com/linode/manager/pull/9156

**Dependabot alert(s)**:
  - https://github.com/advisories/GHSA-f9xv-q969-pqx4

**Jira ticket(s)**:
  - No associated Jira tickets found

---

### **`semver`** pinned version: `^7.5.2`
```
eslint-plugin-react 7.37.4
└── semver 6.3.1
```

**Relevant PR(s)**:
  - https://github.com/linode/manager/pull/9425

**Dependabot alert(s)**:
  - https://github.com/linode/manager/security/dependabot/79

**Jira ticket(s)**:
  - No associated Jira tickets found

---


## Changes  🔄

- Removed 2 package resolutions from `package.json` (`cookie` and `nanoid`)


### Verification steps

- [ ] Fully reset dependencies with `pnpm clean && pnpm install && pnpm bootstrap`
- [ ] Run `pnpm why -r cookie` and `pnpm why -r nanoid` to ensure they're resolving to versions higher than those previously pinned
- [ ] Make sure app works! Build it, look for any errors, check unit tests, etc.

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>

- [x] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [x] All unit tests are passing
- [x TypeScript compilation succeeded without errors
- [x] Code passes all linting rules

</details>

hkhalil-akamai commented on pnpm-lock.yaml:
```diff
@@ -11047,10 +11049,8 @@ snapshots:
 
   isomorphic-fetch@2.2.1:
     dependencies:
-      node-fetch: 2.7.0
+      node-fetch: 1.7.3
```
It looks like `node-fetch` is resolving to 1.7.3 which gives a high-sev vulnerability warning from `pnpm audit` -- we might need to keep this resolution until we get rid of `recompose`.

![Screenshot 2025-04-23 at 5 09 08 PM](https://github.com/user-attachments/assets/17400e55-e92f-494c-98f6-3c37d269d5bf)

bill-akamai replied:
Thanks for catching this! I audited things a bit out of order so updated the change and PR description.

bnussman-akamai commented on package.json:
```diff
@@ -68,13 +68,6 @@
     "prepare": "husky"
   },
-  "resolutions": {
-    "node-fetch": "^2.6.7",
```
Looks like recompose is using `node-fetch` `1.7.3`

Like @hkhalil-akamai said, I think we'll need to keep this resolution until we get rid of `recompose`. (We are close to getting rid of recompose)
 
![Screenshot 2025-04-24 at 9 42 46 AM](https://github.com/user-attachments/assets/6505edb7-9869-4a0f-a1d4-992aa11addd0)

bnussman-akamai commented on package.json:
```diff
@@ -68,13 +68,6 @@
-  "resolutions": {
-    "node-fetch": "^2.6.7",
-    "yaml": "^2.3.0",
-    "semver": "^7.5.2",
```
Looks like `eslint-plugin-react` is using `semver` `6.3.1` 

Maybe we need to keep this resolution for now too?

![Screenshot 2025-04-24 at 9 48 42 AM](https://github.com/user-attachments/assets/4500435b-7403-4add-b159-64539eae501d)

hkhalil-akamai commented:
> After removing the resolutions, I had to update React imports to use the namespace pattern since newer TypeScript types recognize React doesn't have a default export.

I'm not necessarily seeing this error... can others confirm? Totally a matter of preference but `import React` just looks cleaner to me than `import * as React`.


https://github.com/user-attachments/assets/f2308e86-27cc-4432-927e-cef20e953667

bill-akamai commented:
> I'm not necessarily seeing this error... can others confirm? Totally a matter of preference but `import React` just looks cleaner to me than `import * as React`.

@hkhalil-akamai I agree with you that `import React from 'react';` looks cleaner. This style of import fixed some linting issues that showed up after I removed the resolutions however, since it appears  that we need some of these resolutions after all I reverted these changes. It must have stemmed from using the newer version of either `node-fetch`, `yaml`, or `semver`. We can revisit once those are actually ready to be replace if we do need to update the import style then.
